src/components/button/button.scss

```diff
@@ -5,11 +5,14 @@
 @import "./../../style-config/buttons";
 
 .carbon-button {
+  align-items: center;
   border: $carbon-button__border--secondary;
   border-radius: $carbon-button__border-radius;
   box-sizing: border-box;
   cursor: pointer;
-  display: inline-block;
+  display: inline-flex;
```

Comment:
These CSS changes seem to have broken the button contents in Safari.
<img width="154" alt="screen shot 2017-07-12 at 09 09 24" src="https://user-images.githubusercontent.com/9549240/28108009-ed4c8d5c-66e1-11e7-996f-43944a898b86.png">
<img width="178" alt="screen shot 2017-07-12 at 09 09 27" src="https://user-images.githubusercontent.com/9549240/28108010-ed57287a-66e1-11e7-942e-e2215250b783.png">